cstrings/domain.go

```diff
@@ -0,0 +1,19 @@
+package cstrings
+
+import (
+	"regexp"
+	"strings"
+)
+
+// IsValidDomainName is a primitive domain name checker
+// very relaxed and does not conform to spec
+func IsValidDomainName(v string) bool {
```

Comment:
this will be used mostly for SSH host CAs, so I want to be a bit more strict here on purpose, may be I should add a comment about that